WMF Framework/RelatedSearchFetcher.swift

```diff
@@ -60,7 +60,7 @@ final class RelatedSearchFetcher: Fetcher {
     @objc func fetchRelatedArticles(forArticleWithURL articleURL: URL?, resultLimit: Int = RelatedSearchFetcher.MaxResultLimit, completion: @escaping (Error?, [MWKSearchResult]?) -> Void) {
         guard
             let articleURL = articleURL,
-            let articleTitle = articleURL.wmf_titleWithUnderscores
+            let articleTitle = articleURL.wmf_titleWithUnderscores?.addingPercentEncoding(withAllowedCharacters: .wmf_articleTitlePathComponentAllowed)
```

Comment:
Maybe `addingPercentEncoding` should move into `mutating func replacePercentEncodedPathWithPathComponents(_ pathComponents: [String]?) {` to prevent this mistake in the future

Reply:
Good idea 👍

Reply:
Moving this introduces issues with other path components, so parking it for now